package.json

```diff
@@ -129,7 +129,7 @@
     "source-map-support": "^0.5.13",
     "tiny-glob": "^0.2.6",
     "tslib": "^1.10.0",
-    "typescript": "^3.5.3"
+    "typescript": "^4.0.0"
```

Comment:
Is there something that makes upgrading TypeScript to 4.0 necessary? See also https://github.com/sveltejs/svelte/pull/5746 where I proposed this and we decided not to do it for now. The `eslint` plugins have been updated in `master`, so would not need to be bumped here if you rebase against `master`

Reply:
It's been a long time, so I don't remember on top of my head now, but your explanation makes sense. I'll update accordingly.

Reply:
TypeScript 4.0 is not strictly needed, but [3.7+](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) needed for [optional changing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining). I updated it to use `3.9.7` instead. It's minor upgrade, so should not have backward compatibility issue.

Reply:
it looks like the tests are failing on the optional chaining

Reply:
new commit pushed, besides typescript, sucrase also needs to be updated.